programmers/모의고사.js

```diff
@@ -0,0 +1,29 @@
+function solution(answers) {
+   const numberOne = [1,2,3,4,5];
+   const numberTwo = [2,1,2,3,2,4,2,5];
+   const numberThree = [3,3,1,1,2,2,4,4,5,5];
+   let firstCnt = 0;
+   let secondCnt = 0;
+   let thirdCnt = 0;
+
+   for(let i = 0; i < answers.length; i++) {
+     if (numberOne[i % 5] === answers[i]) firstCnt++;
+     if (numberTwo[i % 8] === answers[i]) secondCnt++;
+     if (numberThree[i % 10] === answers[i]) thirdCnt++; 
+   }
+
+   if ((firstCnt > secondCnt) && (firstCnt > thirdCnt))
+        return [1];
+    else if ((firstCnt === secondCnt) && (firstCnt > thirdCnt))
+        return [1, 2];
+    else if ((firstCnt > secondCnt) && (firstCnt === thirdCnt))
+        return [1, 3];
+    else if ((secondCnt > firstCnt) && (secondCnt > thirdCnt))
+        return [2];
+    else if ((secondCnt > firstCnt) && (secondCnt === thirdCnt))
+        return [2, 3];
+    else if ((thirdCnt > firstCnt) && (thirdCnt > secondCnt))
+        return [3];
+    else
+        return [1,2,3];
+} 
```

Comment:
모든 조건을 생각하기 보다 count의 최댓값을 구하고 그에 해당하는 값을 찾아내면 훨씬 가독성이 높음

```suggestion
  const maxCount= Math.max(...count);
    
  count.forEach((el, i) => {
      if (el === maxCount) result.push(i + 1);
  })
    
  return result;
}
```

Reply:
전체 수정 코드
```js
function solution(answers) {
   const patterns = [
       [1, 2, 3, 4, 5],
       [2, 1, 2, 3, 2, 4, 2, 5],
       [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
   ];
   const count = [0, 0, 0];
   const result = [];

   for (let i = 0; i < answers.length; i++) {
     if (patterns[0][i % patterns[0].length] === answers[i]) count[0]++;
     if (patterns[1][i % patterns[1].length] === answers[i]) count[1]++;
     if (patterns[2][i % patterns[2].length] === answers[i]) count[2]++; 
   }

    const maxCount = Math.max(...count);
    
    count.forEach((el, i) => {
        if (el === maxCount ) result.push(i + 1);
    })
    
    return result;
}
```